panther_hardware_interfaces/test/test_panther_imu.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdint>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <hardware_interface/resource_manager.hpp>
+#include <hardware_interface/types/lifecycle_state_names.hpp>
+
+#include <lifecycle_msgs/msg/state.hpp>
+
+#include <panther_utils/test/test_utils.hpp>
+
+#include <panther_imu_test_utils.hpp>
+
+TEST(TestPantherImu, check_interfaces)
```

Comment:
This might be a significant change for these tests, but I suggest that these utilities and constants be organized within a class that inherits from testing::Test, similar to the approach in other packages like panther_battery or panther_lights.

Additionally, there are many functionalities that remain untested, such as exception throwing, calibration, various process failures, and more. It would be beneficial to expand the tests to cover these scenarios.

Reply:
I am concerned that the first part of this comment has not been addressed.